templates/dockerize.mvn.yml

```diff
@@ -0,0 +1,118 @@
+# This template is expects exactly ONE Dockerfile in the microservice repo
+# This template utilizes an external dependency called mvn-version-1.2.1 to 
+# extract the version of the Maven pom.xml file, which is then used as the
+# tag for the docker image that is pushed.
+
+parameters:
+
+  # project configuration parameters:
+- name: imageName                   # <-- e.g., project-registry-gateway
+  type: string
+- name: containerRepo               # <-- e.g., myproject.azurecr.io, myDockerHub
+  type: string
+- name: containerServiceConnection  # <-- name of the service connection associated
+  type: string                      #     with the above repo
+- name: containerUrl                # <-- URL to the associated repo
+  type: string
+- name: version                     # <-- you should only use this parameter if the value of
+  type: string                      #     useDynamicVersion is set to false, otherwise it is 
+  default: undefined                #     overwritten dynamically using mvn-version-1.2.1
+- name: useDynamicVersion           # <-- if this is set to false then it will use the value
+  type: boolean                     #     specified by parameters.version rather than the version
+  default: true                     #     pulled from the pom.xml file (default behavior)
+
+  # external access parameters:
+- name: discordChannel
+  type: string
+- name: discordKey
+  type: string
+
+steps:
+- ${{ if eq(parameters.useDynamicVersion, true) }}:
+  - script: |
+      MVN_VERSION_DOWNLOAD='https://github.com/emcrisostomo/mvn-version/archive/refs/tags/1.2.1.tar.gz'
+      mkdir mvn-version && 
+      cd mvn-version &&
+      mkdir 1.2.1 &&
+      curl -L $MVN_VERSION_DOWNLOAD |
+        tee 1.2.1.tar.gz |
+        tar zx -C 1.2.1
```

Comment:
couple things here, not very important but i typed them
1. you can do `mkdir -p mvn-version/1.2.1` and it'll make both directories in one go if they don't exist
2. do you need to save the file with tee? it's fine, but i'd just pipe directly into tar.
3. tar can create its own output directory with the `--one-top-level` option, though it's more common to do it your way and there's something to be said for that.
4. probably don't need to go two subfolders deep for this. it'd be nice if the number of repetitions of `1.2.1` was lower so it'd be easier in theory to update.

Reply:
streamlined this script and associated file structure per your suggestion